service/tablet_allocator.cc

```diff
@@ -264,6 +264,75 @@ class load_balancer {
         }
     };
 
+    // Maximum tablet size works as the split threshold, whereas minimum size works as the
+    // merge threshold. To prevent oscillations between split and merge, hysteresis is applied
+    // to the decision.
+    // TODO: possibly make target_max_tablet_size a config to help writing tests that stresses resize path.
+    static const size_t target_max_tablet_size = 10UL * 1024 * 1024 * 1024;
+    static const size_t target_min_tablet_size = double(target_max_tablet_size / 2) * 0.6;
```

Comment:
At a minimum this needs to be a configuration item (live updateable). Then you can have a much lower value for tests.

Suggest having a target tablet size of 5GB and then max = 2*target, min = 0.5*target, target = geometric_mean(min, max).

Reply:
Later, we can calculate the target size from the disk space and shard count.

Reply:
I had a discussion with @tgrabiec on this. I also thought we'd have a concept target tablet size, but the idea is to only have maximum (split threshold) and minimum (merge threshold) targets.

Naturally, merge threshold would be 0.5 * (split_threshold / 2), such that after merge we're half way to split or merge. But Tomek proposed delaying merge (hysterisis) to tackle possible oscillations between split and merge. 

I think we can live only with the notion of maximum and derive split (== maximum) and merge thresholds (X * (maximum / 2)) from it. Is there any scenario where you think it might be benefitial to have min,max and target (a point between min and max)?